spec/db/fixtures/ae_datastore/ManageIQ/Cloud/Orchestration/Provisioning/StateMachines/Methods.class/__methods__/check_provisioned_spec.rb

```diff
@@ -0,0 +1,131 @@
+
+  let(:miq_request_task) do
+    FactoryGirl.create(:miq_request_task,
+                       :destination => service_orchestration,
+                       :miq_request => request)
+  end
+
+  let(:amazon_stack) do
+    FactoryGirl.create(:orchestration_stack_amazon)
+  end
+
+  let(:svc_model_orchestration_manager) do
+    MiqAeMethodService::MiqAeServiceExtManagementSystem.find(ems_amazon.id)
+  end
+
+  let(:svc_model_amazon_stack) do
+    MiqAeMethodService::MiqAeServiceOrchestrationStack.find(amazon_stack.id)
+  end
+
+  let(:svc_model_service) do
+    MiqAeMethodService::MiqAeServiceService.find(service_orchestration.id)
+  end
+
+  let(:svc_model_miq_request_task) do
+    MiqAeMethodService::MiqAeServiceMiqRequestTask.find(miq_request_task.id)
+  end
+
+  let(:root_hash) do
+    { 'service_template' => MiqAeMethodService::MiqAeServiceService.find(service_orchestration.id) }
+  end
+  let(:root_object) do
+    obj = Spec::Support::MiqAeMockObject.new(root_hash)
+    obj["service_template_provision_task"] = svc_model_miq_request_task
+    obj
+  end
+  let(:ae_service) do
+    Spec::Support::MiqAeMockService.new(root_object).tap do |service|
+      current_object = Spec::Support::MiqAeMockObject.new
+      current_object.parent = root_object
+      service.object = current_object
+    end
+  end
+
+  before(:each) do
+    allow(svc_model_miq_request_task).to receive(:destination).and_return(svc_model_service)
+  end
+
+  it "waits for the deployment to complete" do
+    allow(svc_model_service).to receive(:orchestration_stack_status) { ['CREATING', nil] }
+    described_class.new(ae_service).main
+
+    expect(ae_service.root['ae_result']).to eq('retry')
+  end
+
+  it "catches the error during stack deployment" do
+    allow(svc_model_service).to receive(:orchestration_stack_status).and_return(['CREATE_FAILED', failure_msg])
+    described_class.new(ae_service).main
+
+    expect(ae_service.root['ae_result']).to eq('error')
+    expect(ae_service.root['ae_reason']).to eq(failure_msg)
+    expect(request.reload.message).to eq(failure_msg)
+  end
+
+  it "truncates the error message that exceeds 255 characters" do
+    allow(svc_model_service).to receive(:orchestration_stack_status).and_return(['CREATE_FAILED', long_failure_msg])
+    described_class.new(ae_service).main
+
+    expect(ae_service.root['ae_result']).to eq('error')
+    expect(ae_service.root['ae_reason']).to eq(long_failure_msg)
+    expect(request.reload.message).to eq('t' * 252 + '...')
+  end
+
+  it "considers rollback as provision error" do
+    allow(svc_model_service)
+      .to receive(:orchestration_stack_status) { ['ROLLBACK_COMPLETE', 'Stack was rolled back'] }
+    described_class.new(ae_service).main
+
+    expect(ae_service.root['ae_result']).to eq('error')
+    expect(ae_service.root['ae_reason']).to eq('Stack was rolled back')
+  end
+
+  it "refreshes the provider and waits for it to complete" do
```

Comment:
Please group all refresh related tests in a context and share the same `before` block where you can prepare for example to return `svc_model_amazon_stack`